CMakeLists.txt

```diff
@@ -15,6 +15,35 @@ message(STATUS "Building the MPACT compiler at ${MPACT_SOURCE_DIR} (into ${MPACT
 
 set(MPACT_PYTHON_PACKAGES_DIR "${MPACT_BINARY_DIR}/python_packages")
 
+#-------------------------------------------------------------------------------
+# Configure out-of-tree vs in-tree build
+#-------------------------------------------------------------------------------
+
+if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+  message(STATUS "Torch-MLIR out-of-tree build.")
+  message(FATAL_ERROR "TODO")
+else()
+  message(STATUS "Torch-MLIR in-tree build.")
+  # In-tree build with LLVM_EXTERNAL_PROJECTS=torch-mlir
+
+  option(MLIR_ENABLE_BINDINGS_PYTHON "Enables MLIR Python Bindings" OFF)
+
+  # TODO: Fix this upstream so that global include directories are not needed.
+  set(MLIR_MAIN_SRC_DIR ${LLVM_MAIN_SRC_DIR}/../mlir)
+  set(MLIR_INCLUDE_DIR ${LLVM_MAIN_SRC_DIR}/../mlir/include)
+  set(MLIR_GENERATED_INCLUDE_DIR ${LLVM_BINARY_DIR}/tools/mlir/include)
+  set(MLIR_INCLUDE_DIRS "${MLIR_INCLUDE_DIR};${MLIR_GENERATED_INCLUDE_DIR}")
+endif()
+
+include_directories(${LLVM_INCLUDE_DIRS})
+include_directories(${MLIR_INCLUDE_DIRS})
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
+include_directories(${CMAKE_CURRENT_BINARY_DIR}/include)
+
+add_subdirectory(include)
```

Comment:
sort add_subdirs alphabetically

Reply:
include -> lib -> tool seems more natural logically, don't you think so?

Reply:
Ah, yeah, that subtlety may be lost in the long run, but I am okay with this order too